### ISSUE TITLE
docker-py v2.1.0 - build context missing files

### ISSUE THREAD

**Issue by jamesongithub** [docker/docker-py#1471]
```
17:21:05 [localhost] local: ls -la ${PATH}/src
17:21:05 drwxr-xr-x  6 ${USER} ${USER} 4096 Feb 20 17:18 .
17:21:05 drwxr-xr-x  8 ${USER} ${USER} 4096 Feb 20 17:19 ..
17:21:05 -rw-r--r--  1 ${USER} ${USER}  751 Feb 20 17:18 403.html
17:21:05 -rw-r--r--  1 ${USER} ${USER}  755 Feb 20 17:18 404.html
17:21:05 -rw-r--r--  1 ${USER} ${USER}  751 Feb 20 17:18 500.html
17:31:27 Sending docker build context...
17:31:27 Traceback (most recent call last):
...
17:31:27   File "/mnt/${USER}/${SITE_PACKAGES_PATH}/site-packages/docker/models/images.py", line 172, in build
17:31:27     raise BuildError(event.get('error') or event)
17:31:27 docker.errors.BuildError: lstat ${PATH}/src/403.html: no such file or directory
```

I reverted to `v2.0.2` and it built fine. Took me a while to determine it was docker-py issue.

**Comment by shin-** [comment]
What does your `.dockerignore` look like?

**Comment by mikz** [comment]
I can report that `**/.*` matches any file with a dot. Which is wrong and different from the official docker client.

**Comment by jamesongithub** [comment]
```
18:37:17     raise docker.errors.DockerException(error.get('message'))
18:37:17 docker.errors.DockerException: COPY failed: stat /mnt/docker/17.06.0~ce-0/tmp/docker-builder014221688/backend/package.json: no such file or directory
18:37:17 Build step 'Execute shell' marked build as failure
```

partial .dockerignore
```
**/.*/
**/coverage/
**/logs/
**/node_modules/
**/test/

**/.*
**/*.crt
**/*.md
**/*.log
**/*.pyc
**/keys.json

*.py
*.sh
*jscs*

```

**Comment by jamesongithub** [comment]
@shin-  occurs in all versions past 2.0.2. 

does looks like the issue @mikz raised.

**Comment by shin-** [comment]
Should have been fixed by #1679 - let me know if that's not the case and I'll reopen.
